packages/back-end/src/api/metrics/deleteMetric.ts

```diff
@@ -14,8 +14,18 @@ export const deleteMetricHandler = createApiRequestHandler(getMetricValidator)(
       throw new Error("Could not find metric with that id");
     }
 
-    if (!req.context.permissions.canDeleteMetric(metric)) {
-      req.context.permissions.throwPermissionError();
+    if (metric.managedBy === "admin") {
+      if (
+        !req.context.permissions.canDeleteOfficialResources({
+          projects: metric.projects,
+        })
+      ) {
+        req.context.permissions.throwPermissionError();
+      }
+    } else {
```

Comment:
I don't think this is an else/if situation. We always want to check `canDeleteMetric`.  If it's managedBy admin, then we want to do an additional check on top of that.

Also, you can just revert all of these changes since you are already doing this check inside of `deleteMetricById`.  Same goes for all the update endpoints.

We currently aren't doing managedBy checks inside of the Model `insertMetric` and `insertMetrics`, but it makes sense to add it there too and keep it consistent.  That way, there will be no changes at all to any controllers or rest endpoints and everything will be self-contained inside of the model.